src/vm/gdbjit.cpp

```diff
@@ -314,75 +311,42 @@ void NotifyGdb::MethodCompiled(MethodDesc* MethodDescPtr)
     /* Build .debug_str section */
     if (!BuildDebugStrings(dbgStr))
     {
-        delete[] symInfo;
-        delete[] dbgAbbrev.MemPtr;
-        delete[] dbgLine.MemPtr;
         return;
     }
     
     /* Build .debug_info section */
     if (!BuildDebugInfo(dbgInfo))
     {
-        delete[] symInfo;
-        delete[] dbgAbbrev.MemPtr;
-        delete[] dbgLine.MemPtr;
-        delete[] dbgStr.MemPtr;
         return;
     }
     
     /* Build .debug_pubname section */
     if (!BuildDebugPub(dbgPubname, methodName, dbgInfo.MemSize, 26))
     {
-        delete[] symInfo;
-        delete[] dbgAbbrev.MemPtr;
-        delete[] dbgLine.MemPtr;
-        delete[] dbgStr.MemPtr;
-        delete[] dbgInfo.MemPtr;
         return;
     }
     
     /* Build debug_pubtype section */
     if (!BuildDebugPub(dbgPubType, "int", dbgInfo.MemSize, 37))
     {
-        delete[] symInfo;
-        delete[] dbgAbbrev.MemPtr;
-        delete[] dbgLine.MemPtr;
-        delete[] dbgStr.MemPtr;
-        delete[] dbgInfo.MemPtr;
-        delete[] dbgPubname.MemPtr;
         return;
     }
     
     /* Build section names section */
     if (!BuildSectionNameTable(sectStr))
     {
-        delete[] symInfo;
-        delete[] dbgAbbrev.MemPtr;
-        delete[] dbgLine.MemPtr;
-        delete[] dbgStr.MemPtr;
-        delete[] dbgInfo.MemPtr;
-        delete[] dbgPubname.MemPtr;
-        delete[] dbgPubType.MemPtr;
         return;
     }
 
     /* Build section headers table */
     if (!BuildSectionTable(sectHeaders))
     {
-        delete[] symInfo;
-        delete[] dbgAbbrev.MemPtr;
-        delete[] dbgLine.MemPtr;
-        delete[] dbgStr.MemPtr;
-        delete[] dbgInfo.MemPtr;
-        delete[] dbgPubname.MemPtr;
-        delete[] dbgPubType.MemPtr;
-        delete[] sectStr.MemPtr;
         return;
     }
 
     /* Patch section offsets & sizes */
     long offset = sizeof(Elf_Ehdr);
-    Elf_Shdr* pShdr = reinterpret_cast<Elf_Shdr*>(sectHeaders.MemPtr);
+    Elf_Shdr* pShdr = reinterpret_cast<Elf_Shdr*>(sectHeaders.MemPtr.Extract());
```

Comment:
It seems you really want GetValue instead of Extract (and the same at all other places). The Extract method suppresses release of the pointer, so the memory would not be freed by the destructor, while the GetValue just gets the value of the pointer.